src/helpers/LabelFilterHelper.ts

```diff
@@ -4,36 +4,91 @@ import { ServiceListItem } from '../types/ServiceList';
 
 type itemsType = AppListItem | ServiceListItem | WorkloadListItem;
 
-export const filterByLabel = (items: itemsType[], filter: string[]): itemsType[] => {
-  let result: itemsType[] = [];
-  filter.map(filter => {
-    if (filter.includes(':')) {
-      const values = filter.split(':');
-      // Check Values
-      values[1]
-        .split(',')
-        .map(
-          val =>
-            (result = result.concat(
-              items.filter(
-                item =>
-                  values[0] in item.labels &&
-                  item.labels[values[0].trim()]
-                    .split(',')
-                    .some(appVal => appVal.trim().startsWith(val.trim()) && !result.includes(item))
-              )
-            ))
-        );
-    } else {
-      // Check if has Label
-      result = result.concat(
-        items.filter(item =>
-          Object.keys(item.labels).some(key => key.startsWith(filter.trim()) && !result.includes(item))
-        )
-      );
+const filterLabelByOp = (labels: { [key: string]: string }, filters: string[], op: string = 'or'): boolean => {
```

Comment:
I think a couple of unit tests for this function and/or `filterByLabel` would be really nice